controllers/cpms/cpms_controller.go

```diff
@@ -60,10 +62,34 @@ func (r *CPMSReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.
 	}
 
 	reqLogger.Info("Found ControlPlaneMachineSet")
+
+	// Fetch the instance type from cpms spec
+	specRaw := cpms.Spec.Template.OpenShiftMachineV1Beta1Machine.Spec.ProviderSpec.Value.Raw
+	var instance_type string
+
+	// the machine template is provider specific
+	if cpms.Spec.Template.OpenShiftMachineV1Beta1Machine.FailureDomains.Platform == "AWS" {
+		machineProviderConfig := machinev1beta1.AWSMachineProviderConfig{}
+		err := json.Unmarshal(specRaw, &machineProviderConfig)
+		if err != nil {
+			reqLogger.Error(err, "failed to unmarshal machine config")
+			return utils.RequeueWithError(err)
+		}
+		instance_type = machineProviderConfig.InstanceType
+	} else {
+		machineProviderConfig := machinev1beta1.GCPMachineProviderSpec{}
```

Comment:
CPMS in general supports way more cloud providers than GCP and AWS, so I would put this block in an explicit GCP check and error out in case this operator is running on an unsupported cloud provider.

Reply:
Thats probably a good idea :D I made it explicit now. 
I was not thinking about the other cloud providers, as the metrics exporter is intended to only be run on OSD.